compiler/rustc_macros/src/query.rs

```diff
@@ -555,14 +558,24 @@ pub fn rustc_queries(input: TokenStream) -> TokenStream {
                 }
             }
         }
+        #[macro_export]
         macro_rules! rustc_cached_queries {
             ($($macro:tt)*) => {
                 $($macro)*(#cached_queries);
             }
         }
 
-        #query_description_stream
+        #[macro_export]
+        macro_rules! query_description_stream {
+            // capture all needed `use` statements.
+            ($($uses:tt)*) => {
+                $($uses)*
```

Comment:
You can surround thins expansion by an ad-hoc module to avoid polluting the namespace.